src/main/java/com/google/devtools/common/options/Converters.java

```diff
@@ -625,4 +635,73 @@ public String getTypeDescription() {
       return "Converts to a CacheBuilderSpec, or null if the input is empty";
     }
   }
+
+  public static class ExpressionConverter implements Converter<Expression> {
+    @Override
+    public Expression convert(String input) throws OptionsParsingException {
+      if (input.isEmpty()) {
+        return Expression.empty();
+      }
+      if (isLegacySyntax(input)) {
+        input = convertLegacySyntaxToBooleanExpression(input);
+      }
+      try {
+        Expression result = Expression.parse(ParserInput.fromLines(input));
+        checkForForbiddenOperators(result);
+        return result;
+      } catch (SyntaxError.Exception e) {
+        throw new OptionsParsingException("Failed to parse expression: " + e.getMessage() + " input: " + input, e);
+      }
+    }
+
+    @Override
+    public String getTypeDescription() {
+      return "Converts to an Expression.";
```

Comment:
This string should be a noun-phrase that concisely describes what the user is expected to type. For example:

"A Boolean formula over test tags (e.g. '(release or smoke) and not flaky')".